nautobot/core/filters.py

```diff
@@ -501,7 +501,7 @@ def generate_query(self, value, qs=None, **kwargs):
         predicates = []
         for obj in value:
             # Try to get the `to_field_name` (e.g. `name`) or just pass the object through.
-            val = getattr(obj, self.field.to_field_name, obj)
```

Comment:
For my own edification, how is the filter still able to filter on `name` if we make this change here?

Reply:
@glennmatthews Might be able to point to the line where this is happening but by the time we get to `generate_query` we already have actual instances (hence the `node.descendants(include_self=True)` above). When you intend to search for something by name, you want to find all instances that match that name. Without this fix, it is using the name of the descendants to then perform the query again (find all instances that match the name of my descendants) you expand the resulting query.

Something like:

```
Blues
  Teal
  Aquamarine
  Sky
Reds
  Crimson
  Strawberry
  Burgundy
```

If you search for "Devices in Locations with the name 'Red'", the resulting filter becomes "Devices in Locations with the names ['Red', 'Crimson', 'Strawberry', 'Burgundy']". Fine but if you add "Purple" as a Location in both Blues and Reds, Devices in Blues > Purple would show up too.

Instead what is performed is "Give me all Locations with the name 'Red'" and then to process the tree, it then performs "Give me all Locations matching this PK which are the descendants of Locations with the name 'Red'".